### PR TITLE
Fix Travis deploy condition

### PR DESCRIPTION
**Pull request by cduck**
Follow up to #1938.

### CODE REVIEW

**Comment by dabacon** [comment]
For some reason Travis did not pick this up.

**Comment by cduck** [comment]
I moved set-dev-version.sh to it's own Travis job because I don't think before deploy will run unless the condition is already met.  (https://docs.travis-ci.com/user/deployment/pypi/#running-commands-before-and-after-release)  This has the benefit that every job won't try push its own dev release (each with a different timestamp).

It seems that `*` characters need to be quoted otherwise Travis doesn't run.

@dabacon PTAL.

**Comment by cduck** [comment]
I think the condition is working now.

File: .travis.yml [hunk 1]
```diff
@@ -60,22 +70,28 @@ matrix:
...
     env: NAME=doctest
     python: '3.6'
     install:
     - pip install -r requirements.txt
     - pip install -r cirq/contrib/contrib-requirements.txt
     script: check/doctest -q
+  - stage: deploy
+    os: linux
+    env: NAME=dev-release
+    python: '3.7'
```
Comment (dabacon) [inline .travis.yml]: why 3.7?  I think we say we support python 3.6

**Comment by dabacon** [comment]
One comment about python version.

File: .travis.yml (hunk 1 above)
Comment (cduck) [reply .travis.yml]: I just picked one.  This would just be the python version that runs setup.py.

Comment (dabacon) [reply .travis.yml]: ah k

**Comment by cduck** [comment]
Deploy ran but failed:
```
Uploading distributions to https://test.pypi.org/simple/

Uploading cirq-0.6.0.dev0-py3-none-any.whl

100%|████████████████████████████████████████| 851k/851k [00:00<00:00, 2.27MB/s]

NOTE: Try --verbose to see response content.

HTTPError: 405 Client Error: Method Not Allowed for url: https://test.pypi.org/simple/

No stash entries found.

PyPI upload failed.

failed to deploy
```

**Pull request merged**